### PR TITLE
[MULTIARCH-3486] IBM Z add sno support

### PR DESCRIPTION
**Pull request by SNiemann15**
Version(s): 4.14 + 

Issue: https://issues.redhat.com/browse/MULTIARCH-3486

Link to docs preview:
- [Preparing to install on a single node](https://63974--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_sno/install-sno-preparing-to-install-sno)
- [Installing single-node OpenShift with IBM Z and IBM® LinuxONE](https://63974--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_sno/install-sno-installing-sno#installing-single-node-openshift-with-ibm-z-and-ibm-linuxone)

QE review:
QE: @holgwolf SME: @chidanandbh 
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by chidanandbh** [comment]
[Create parameter files.docx](https://github.com/openshift/openshift-docs/files/12473803/Create.parameter.files.docx)

Update some point in this document for parm file zVM

**Comment by chidanandbh** [comment]
Step 14b:   need to change DASD to FCP



Step14a: can we make it little more generic by providing general name I mean

	#cp i <devno> clear loadparm prompt

		<devno>
		specifies the device number of the boot device as seen by the guest.
	
	Ex: cp i 4411 clear loadparm prompt 


	
	#cp vi vmsg 0 <Kernel Parameter>

	Ex: cp vi vmsg 0 rd.dasd=0.0.4411 ip=encbdd0:dhcp::02:00:00:02:34:02 rd.neednet=1 console=ttysclp0 


Step14b: can we make it little more generic by providing general name I mean

	1) #cp set loaddev portname <wwpn> lun <lun>

	   ex: cp set loaddev portname 50050763 0a0350a4 lun 4000409D 00000000


	2) #cp set loaddev bootprog <n>
		where <n> selects the kernel to be booted.
	   Ex: cp set loaddev bootprog 0

	
	3) Add kernel parameters

	    #cp set loaddev scpdata <APPEND|NEW> '<kernel_parameters>'

			<kernel_parameters>
				specifies a set of kernel parameters to be stored as system control program data (SCPDATA). When booting Linux, these kernel 
				parameters are concatenated to the end of the existing kernel parameters that are used by your boot configuration. The 
				combined parameter string must not exceed 896 characters.

			Optional: APPEND
			appends kernel parameters to existing SCPDATA. This is the default.

			Optional: NEW
			replaces existing SCPDATA.


	    ex: cp set loaddev scpdata 'rd.zfcp=0.0.8001,0x500507630a0350a4,0x4000409D00000000 ip=encbdd0:dhcp::02:00:00:02:34:02 rd.neednet=1'



	4) Start the IPL and boot process by entering a command of this form
	
	    #cp i <devno> 

	    Ex: cp i 8001

**Comment by chidanandbh** [comment]
For Reference some docs to refer

https://www.ibm.com/docs/en/linux-on-systems?topic=blz-from-dasd-1
https://www.ibm.com/docs/en/linux-on-systems?topic=blz-from-scsi-1

**Comment by chidanandbh** [comment]
https://63974--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_sno/install-sno-installing-sno#installing-sno-on-ibm-z_install-sno-installing-sno-with-the-assisted-installer

**Installing single-node OpenShift with z/VM on IBM Z and IBM® LinuxONE**
Step 6  and Step 15 are identical , please remove Step 15

**Comment by chidanandbh** [comment]
https://63974--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_sno/install-sno-installing-sno#installing-sno-on-ibm-z-kvm_install-sno-installing-sno-with-the-assisted-installer

**Installing single-node OpenShift with RHEL KVM on IBM Z and IBM® LinuxONE**
Step 6 and Step 10 are identical , please remove Step 10

**Comment by SNiemann15** [comment]
@alishaIBM FYI I've added Power to the Preparing SNO section.

File: modules/install-sno-requirements-for-installing-on-a-single-node.adoc [hunk 1]
```diff
@@ -10,17 +10,22 @@ Installing {product-title} on a single node alleviates some of the requirements
...
-* *CPU Architecture:* Installing {product-title} on a single node supports `x86_64` and `arm64` CPU architectures.
+[NOTE]
+====
+ISO is not required for {ibmzProductName} installations.
+====
+
+* *CPU Architecture:* Installing {product-title} on a single node supports `x86_64`, `arm64`,`ppc64le`, and `s390x` CPU architectures.
 
-* *Supported platforms:* Installing {product-title} on a single node is supported on bare metal, vSphere, AWS, Red Hat OpenStack, and {rh-virtualization-first} platforms.
+* *Supported platforms:* Installing {product-title} on a single node is supported on bare metal, vSphere, AWS, Red Hat OpenStack, {rh-virtualization-first}, {ibmpowerProductName}, and{ibmzProductName} platforms.
```
Comment (SNiemann15) [inline modules/install-sno-requirements-for-installing-on-a-single-node.adoc:20]: @emarcusRH @apinnick  {rh-virtualization-first} didn't resolve in my preview and I assume the entry got missed when you were working on this effort to remove RHV docs: https://issues.redhat.com/browse/OCPRHV-773

Can I simply remove the entry in my PR? Or change to hardcoded?

Comment (apinnick) [reply modules/install-sno-requirements-for-installing-on-a-single-node.adoc:20]: This looks like an error. AFAIK, we have never supported RHV on SNO. I think that should be `{VirtProductName}`.

File: installing/installing_sno/install-sno-installing-sno.adoc [hunk 2]
```diff
@@ -75,3 +75,29 @@ include::modules/install-sno-installing-with-usb-media.adoc[leveloffset=+1]
 include::modules/install-booting-from-an-iso-over-http-redfish.adoc[leveloffset=+1]
 
 include::modules/creating-custom-live-rhcos-iso.adoc[leveloffset=+1]
+
+== Installing {sno} with {ibmzProductName} and {linuxoneProductName}
+
+Installing a single node cluster on {ibmzProductName} and {linuxoneProductName} requires user-provisioned installation using either the "Installing a cluster with {op-system-base} KVM on {ibmzProductName} and {linuxoneProductName}" or the "Installing a cluster with z/VM on {ibmzProductName} and {linuxoneProductName}" procedure.
+
+[discrete]
+=== Hardware requirements
```
Comment (SNiemann15) [inline installing/installing_sno/install-sno-installing-sno.adoc:84]: @chidanandbh Can you please verify if this is correct. @holgwolf asked me to add Hardware Requirements

File: modules/install-sno-ibm-z-kvm.adoc [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+to the {product-title} version that you install.Only use the appropriate `kernel`, `initramfs`, and `rootfs` artifacts described in the following procedure.
+====
++
+The file names contain the {product-title} version number. They resemble the following examples:
+
+** kernel: `rhcos-<version>-live-kernel-<architecture>`
+** initramfs: `rhcos-<version>-live-initramfs.<architecture>.img`
+** rootfs: `rhcos-<version>-live-rootfs.<architecture>.img`
++
+. Move the downloaded {op-system-base} live `kernel`, `initramfs`, and `rootfs` as well as the Ignition files to an HTTP or HTTPS server before you launch `virt-install`.
```
Comment (michaelryanpeter) [inline modules/install-sno-ibm-z-kvm.adoc:139]:
```suggestion
. Move the downloaded {op-system-base} live `kernel`, `initramfs`, and `rootfs` artifacts as well as the Ignition files to an HTTP or HTTPS server before you launch `virt-install`.
```

File: modules/install-sno-ibm-z-kvm.adoc [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+   --extra-args "ip=dhcp rd.neednet=1 ignition.platform.id=metal ignition.firstboot" \
+   --extra-args "coreos.live.rootfs_url={rhcos_liveos}" \ <2>
+   --extra-args "ignition.config.url={rhcos_ign}" \ <3>
+   --extra-args "random.trust_cpu=on rd.luks.options=discard" \
+   --extra-args "console=tty1 console=ttyS1,115200n8" \
+   --wait
+----
+<1> For `--location`, specify the location of the kernel/initrd on the HTTP or HTTPS server.
+<2> For `coreos.live.rootfs_url=`, specify the matching rootfs artifact for the `kernel` and `initramfs` you are booting. Only HTTP and HTTPS protocols are supported.
+<3> For `ignition.config.url=`, specify the Ignition file for the machine role. Only HTTP and HTTPS protocols are supported.
```
Comment (michaelryanpeter) [inline modules/install-sno-ibm-z-kvm.adoc:167]: The literals need to be followed by a noun, such as the `--location` flag, `kernel` and `initramfs` artifacts, or `coreos.live.rootfs_url=` parameter.

File: modules/install-sno-ibm-z.adoc [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+zfcp.allow_lun_scan=0 \
+rd.luks.options=discard \
+ignition.firstboot ignition.platform.id=metal \
+console=tty1 console=ttyS1,115200n8
+----
+<1> For `coreos.live.rootfs_url=`, specify the matching rootfs artifact for the `kernel`and `initramfs` you are booting. Only HTTP and HTTPS protocols are supported.
+<2> For `ignition.config.url=`, specify the Ignition file for the machine role. Only HTTP and HTTPS protocols are supported.
+<3> For ip=, assign the IP address automatically using DHCP or manually as described in "Installing a cluster with z/VM on {ibmzProductName} and {linuxoneProductName}".
+<4> For installations on DASD-type disks, use `rd.dasd=` to specify the DASD where {op-system} is to be installed. Omit this entry for FCP-type disks.
+<5> For installations on FCP-type disks, use `rd.zfcp=<adapter>,<wwpn>,<lun>` to specify the FCP disk where {op-system} is to be installed. Omit this entry for DASD-type disks.
```
Comment (michaelryanpeter) [inline modules/install-sno-ibm-z.adoc:169]: See the comment about about following the literals with a noun.

File: modules/install-sno-ibm-z.adoc [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+rd.dasd=0.0.4411 \ <4>
+rd.zfcp=0.0.8001,0x50050763040051e3,0x4000406300000000 \ <5>
+zfcp.allow_lun_scan=0 \
+rd.luks.options=discard \
+ignition.firstboot ignition.platform.id=metal \
+console=tty1 console=ttyS1,115200n8
+----
+<1> For `coreos.live.rootfs_url=`, specify the matching rootfs artifact for the `kernel`and `initramfs` you are booting. Only HTTP and HTTPS protocols are supported.
+<2> For `ignition.config.url=`, specify the Ignition file for the machine role. Only HTTP and HTTPS protocols are supported.
+<3> For ip=, assign the IP address automatically using DHCP or manually as described in "Installing a cluster with z/VM on {ibmzProductName} and {linuxoneProductName}".
```
Comment (michaelryanpeter) [inline modules/install-sno-ibm-z.adoc:167]:
```suggestion
<3> For `ip=` parameter, assign the IP address automatically using DHCP or manually as described in "Installing a cluster with z/VM on {ibmzProductName} and {linuxoneProductName}".
```

File: modules/install-sno-ibm-z.adoc [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+** kernel: `rhcos-<version>-live-kernel-<architecture>`
+** initramfs: `rhcos-<version>-live-initramfs.<architecture>.img`
+** rootfs: `rhcos-<version>-live-rootfs.<architecture>.img`
++
+[NOTE]
+====
+The rootfs image is the same for FCP and DASD.
+====
+
+. Move the downloaded {op-system-base} live `kernel`, `initramfs`, and `rootfs` as well as the Ignition files to an HTTP or HTTPS server.
```
Comment (michaelryanpeter) [inline modules/install-sno-ibm-z.adoc:144]:
```suggestion
. Move the downloaded {op-system-base} live `kernel`, `initramfs`, and `rootfs` (images/artifacts/files) as well as the Ignition files to an HTTP or HTTPS server.
```

File: modules/install-sno-ibm-z.adoc [hunk 8]
```diff
@@ -0,0 +1,242 @@
...
+----
+<1> For `coreos.live.rootfs_url=`, specify the matching rootfs artifact for the `kernel`and `initramfs` you are booting. Only HTTP and HTTPS protocols are supported.
+<2> For `ignition.config.url=`, specify the Ignition file for the machine role. Only HTTP and HTTPS protocols are supported.
+<3> For ip=, assign the IP address automatically using DHCP or manually as described in "Installing a cluster with z/VM on {ibmzProductName} and {linuxoneProductName}".
+<4> For installations on DASD-type disks, use `rd.dasd=` to specify the DASD where {op-system} is to be installed. Omit this entry for FCP-type disks.
+<5> For installations on FCP-type disks, use `rd.zfcp=<adapter>,<wwpn>,<lun>` to specify the FCP disk where {op-system} is to be installed. Omit this entry for DASD-type disks.
++
+Leave all other parameters unchanged.
+
+. Transfer the `initramfs`, `kernel`, parameter files, and {op-system} images to z/VM, for example by using FTP. For details about how to transfer the files with FTP and boot from the virtual reader, see link:https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/installation_guide/sect-installing-zvm-s390[Installing under Z/VM].
```
Comment (michaelryanpeter) [inline modules/install-sno-ibm-z.adoc:173]:
```suggestion
. Transfer the `initramfs` file, `kernel` file, parameter files, and {op-system} images to z/VM, for example by using FTP. For details about how to transfer the files with FTP and boot from the virtual reader, see link:https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/installation_guide/sect-installing-zvm-s390[Installing under Z/VM].
```

File: modules/install-sno-ibm-z.adoc [hunk 9]
```diff
@@ -0,0 +1,242 @@
...
+. After firt reboot of the virtual machine, run the following commands directly after one another:
+
+.. To boot a DASD device after first reboot, run the following commands:
++
+[source,terminal]
+----
+$ cp i <devno> clear loadparm prompt
+----
++
+Where `<devno>` specifies the device number of the boot device as seen by the guest.
```
Comment (michaelryanpeter) [inline modules/install-sno-ibm-z.adoc:194]:
```suggestion
[source,terminal]
----
$ cp i <devno> clear loadparm prompt
----
+
where:

`<devno>`:: Specifies the device number of the boot device as seen by the guest.
```

Comment (michaelryanpeter) [reply modules/install-sno-ibm-z.adoc:194]: See the repo guidelines about [Code formatting and syntax](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#code-blocks-command-syntax-and-example-output). You have to scroll a bit, or just search the page for "where:"

File: modules/install-sno-ibm-z.adoc [hunk 10]
```diff
@@ -0,0 +1,242 @@
...
+Where `<n>` specifies the kernel to be booted.
++
+[source,terminal]
+----
+$ cp set loaddev scpdata <APPEND|NEW> '<kernel_parameters>'
+----
++
+Where `<kernel_parameters>` specifies a set of kernel parameters to be stored as system control program data (SCPDATA). When booting Linux, these kernel parameters are concatenated to the end of the existing kernel parameters that are used by your boot configuration. The combined parameter string must not exceed 896 characters.
++
+Optional: <APPEND|NEW> Specify `APPEND` to append kernel parameters to existing SCPDATA. This is the default. Specify `NEW` to replace existing SCPDATA.
```
Comment (michaelryanpeter) [inline modules/install-sno-ibm-z.adoc:226]: I think that it makes sense to have `<APPEND|NEW>` as a second definition term.

```
+
where:

`<kernel_parameters>`:: Specifies...
`<APPEND|NEW>`:: Optional: Specify `APPEND` to...

File: modules/install-sno-ibm-z.adoc [hunk 11]
```diff
@@ -0,0 +1,242 @@
...
+[source,terminal]
+----
+$ cp set loaddev bootprog <n>
+----
++
+Where `<n>` specifies the kernel to be booted.
++
+[source,terminal]
+----
+$ cp set loaddev scpdata <APPEND|NEW> '<kernel_parameters>'
```
Comment (michaelryanpeter) [inline modules/install-sno-ibm-z.adoc:221]:
```suggestion
$ cp set loaddev scpdata [<APPEND|NEW>] '<kernel_parameters>'
```
Based on the [ISG](https://www.ibm.com/docs/en/ibm-style?topic=elements-command-syntax#using-text-to-specify-command-syntax), I think that APPEND|NEW should be wrapped in brackets.

File: modules/install-sno-ibm-z.adoc [hunk 12]
```diff
@@ -0,0 +1,242 @@
...
+
+. Log in to CMS on the bootstrap machine.
+
+. IPL the bootstrap machine from the reader by running the following command:
++
+----
+$ cp ipl c
+----
+
+. After firt reboot of the virtual machine, run the following commands directly after one another:
```
Comment (michaelryanpeter) [inline modules/install-sno-ibm-z.adoc:185]:
```suggestion
. After the first reboot of the virtual machine, run the following commands directly after one another:
```

**Review by michaelryanpeter — COMMENTED** [review]
Great job! Please reach out if you have questions about any of my feedback. 

In my suggestions, I wasn't sure when to use the terms image, artifact, and file when asking for a noun to follow `kernel`, `initramfs`, and `rootfs`. So, my apologies if my suggestions there are inconsistent.

File: modules/install-sno-ibm-z.adoc (hunk 11 above)
Comment (SNiemann15) [reply modules/install-sno-ibm-z.adoc:221]: I think it should be braces, unless I got the ISG wrong here: 

Use braces to enclose a choice of values.
Correct example The syntax is as follows:
docker exec -it Db2wh manage_hadr {-start_as {primary|standby}|-stop|-status|-enable_acr [ssl]|-update_db}

File: modules/install-sno-about-installing-on-a-single-node.adoc [hunk 13]
```diff
@@ -6,7 +6,12 @@
...
 = About OpenShift on a single node
 
-You can create a single-node cluster with standard installation methods. {product-title} on a single node is a specialized installation that requires the creation of a special ignition configuration ISO. The primary use case is for edge computing workloads, including intermittent connectivity, portable clouds, and 5G radio access networks (RAN) close to a base station. The major tradeoff with an installation on a single node is the lack of high availability.
+You can create a single-node cluster with standard installation methods. {product-title} on a single node is a specialized installation that requires the creation of a special Ignition configuration file. The primary use case is for edge computing workloads, including intermittent connectivity, portable clouds, and 5G radio access networks (RAN) close to a base station. The major tradeoff with an installation on a single node is the lack of high availability.
+
+[NOTE]
+====
+Installing a single-node cluster on {ibmzProductName} simplifies installation for development and test environments and requires less resource requirements at entry level.
+====
```
[the remark this answers is not in the document]
Comment (SNiemann15) [reply modules/install-sno-about-installing-on-a-single-node.adoc:14]: Discussed with @holgwolf to move the note to the IBM Z SNO section for now. 
After GA follow up with Power, ARM, and x86 QE if this applies to all arches.

File: modules/install-sno-ibm-z-kvm.adoc [hunk 14]
```diff
@@ -0,0 +1,173 @@
...
+====
++
+The file names contain the {product-title} version number. They resemble the following examples:
++
+`kernel`:: `rhcos-<version>-live-kernel-<architecture>`
+`initramfs`:: `rhcos-<version>-live-initramfs.<architecture>.img`
+`rootfs`:: `rhcos-<version>-live-rootfs.<architecture>.img`
++
+. Before you launch `virt-install`, move the following files and artifacts to an HTTP or HTTPS server:
+
```
Comment (michaelryanpeter) [inline modules/install-sno-ibm-z-kvm.adoc:138]: Nit: If you use a `+` here, you can just use a single `*` for the bullet to get the correct alignment. Not a big deal at all, just an FYI.

File: modules/install-sno-ibm-z-kvm.adoc [hunk 15]
```diff
@@ -0,0 +1,173 @@
...
+** Downloaded {op-system-base} live `kernel`, `initramfs`, and `rootfs` artifacts
+** Ignition files
+
+. Create the KVM guest nodes by using the following components:
+
```
Comment (michaelryanpeter) [inline modules/install-sno-ibm-z-kvm.adoc:143]: Same here.

File: modules/install-sno-ibm-z.adoc [hunk 16]
```diff
@@ -0,0 +1,264 @@
...
++
+--
+[source,terminal]
+----
+$ cp set loaddev portname <wwpn> lun <lun>
+----
+
+where:
+
+`<wwpn>`:: Specifies the target port and `<lun>` the logical unit in hexadecimal format.
```
Comment (michaelryanpeter) [inline modules/install-sno-ibm-z.adoc:225]: Nit/follow up:
```suggestion
`<wwpn>`:: Specifies the target port.
`<lun>`:: Specifies the logical unit in hexadecimal format.
```

**Pull request merged**